src/UI/Component/Input/Field/TagInput.php

```diff
@@ -0,0 +1,134 @@
+<?php
+declare(strict_types=1);
+
+namespace ILIAS\UI\Component\Input\Field;
+
+use ILIAS\UI\Component\JavaScriptBindable;
+use ILIAS\UI\Component\Signal;
+
+/**
+ * Interface TagInput
+ *
+ * This describes Tag Inputs
+ *
+ * @package ILIAS\UI\Component\Input\Field
+ */
+interface TagInput extends Input, JavaScriptBindable {
+
+	const EVENT_ITEM_ADDED = 'itemAdded';
+	const EVENT_BEFORE_ITEM_REMOVE = 'beforeItemRemove';
+	const EVENT_BEFORE_ITEM_ADD = 'beforeItemAdd';
+	const EVENT_ITEM_REMOVED = 'itemRemoved';
+
+
+	/**
+	 * @return array of tags such as [ 'Interesting', 'Boring', 'Animating', 'Repetitious' ]
+	 */
+	public function getTags(): array;
+
+
+	/**
+	 * Get an input like this, but decide whether the user can provide own
+	 * tags or not. (Default: Allowed)
+	 *
+	 * @param bool $extendable
+	 *
+	 * @return \ILIAS\UI\Component\Input\Field\TagInput
+	 */
+	public function withUserCreatedTagsAllowed(bool $extendable): TagInput;
+
+
+	/**
+	 * @see withUserCreatedTagsAllowed
```

Comment:
Cool, like the use of @see, makes sense not to duplicate methods descriptions...